common/common-rest/src/test/java/io/servicecomb/common/rest/TestRestProducerInvocation.java

```diff
@@ -202,6 +202,29 @@ void invoke() {
     Assert.assertTrue(invokeNoParam);
     Assert.assertSame(args, restProducerInvocation.invocation.getSwaggerArguments());
   }
+  
+  @Test
+  public void runOnExecutorException() {
+    new Expectations(RestCodec.class) {
+      {
+        RestCodec.restToArgs(requestEx, restOperationMeta);
+        result = new Exception();
+      }
+    };
+    restProducerInvocation = new MockUp<RestProducerInvocation>() {
+      @Mock
+      void invoke() {
+        invokeNoParam = true;
+      }
+    }.getMockInstance();
+    initRestProducerInvocation();
+    try {
+      restProducerInvocation.runOnExecutor();
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
If the exception is thrown, you need to write assert for the exception and don't just print it out.

Reply:
Assert is done above the expectation, method invoke is not executed and result is false. I think it's his logic.

Reply:
yes, this is simulate a exception

Reply:
best practice is use org.junit.rules.ExpectedException